### DISCUSSION TITLE
How to install juiceFS (server) on Kubernetes?

### DISCUSSION THREAD

**Discussion by guettli** [juicedata/juicefs#4820] [General]
Maybe I am blind, but I found no docs about installing juiceFS (server) on Kubernetes.

Or is that not supported, and only the client is open source?

**Comment by ak2k** [comment]
I happened to see this at one point, so sharing in case helpful: https://juicefs.com/docs/csi/introduction

**Reply by guettli** [reply]
@ak2k thank you for your reply. But maybe I am missing something: The link you mentioned documents the **usage** of juiceFS via CSI.

My question is about: How to deploy juiceFS in my Kubernetes cluster. Only after that step I can actually use juiceFS via CSi.

**Reply by ak2k** [reply]
Hi @guettli, Juicefs accesses configured (external to juicefs) object storage and metadata/"database" storage for persistence (the "server" equivalent).  Hence, you can leverage a public cloud service or self-host compatible equivalents, though self-hosting these enabling services isn't within the intended scope of juicefs.

https://juicefs.com/docs/community/introduction/

**Reply by guettli** [reply]
thank you for your reply. 

Now I understand. You only need the CSI driver. Then you configure s3 and metadata-server, and then you have all you need.
